letsencrypt-apache/letsencrypt_apache/obj.py

```diff
@@ -199,12 +199,17 @@ def same_server(self, vhost):
 
         """
 
-        if vhost.get_names() != self.get_names():
-            return False
+        if not generic:
```

Comment:
I think we can make this logic a little simpler. I think this line should be `if generic and self.name:`